game-core/src/main/java/games/strategy/engine/history/change/units/KillUnits.java

```diff
@@ -0,0 +1,125 @@
+package games.strategy.engine.history.change.units;
+
+import games.strategy.engine.data.CompositeChange;
+import games.strategy.engine.data.Territory;
+import games.strategy.engine.data.Unit;
+import games.strategy.engine.data.changefactory.ChangeFactory;
+import games.strategy.engine.delegate.IDelegateBridge;
+import games.strategy.engine.history.change.HistoryChange;
+import games.strategy.triplea.formatter.MyFormatter;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+import lombok.Value;
+import org.apache.commons.text.StringSubstitutor;
+import org.triplea.java.collections.IntegerMap;
+
+/**
+ * Kills a set of units in a location
+ *
+ * <p>Adds a history event for the killing
+ *
+ * <p>Transforms units to other units if needed. See {@link TransformUnits}
+ */
+@Value
```

Comment:
Is the 'allArgsConstructor' of '@Value' being used?

As-is, it looks like there are a lot of constructors for this object, enough so it might be hard to reason about how it is instantiated, which makes reasoning about the object state difficult and depend on tracing down call flows.

If we do not use the 'all args constructor', removing that can help the above problem and just replace this with 'getter, setter, and equalsAndHashcode'.